lib/commonFunctions.js

```diff
@@ -19,6 +19,42 @@ var sundial = require('sundial');
 
 var annotate = require('./eventAnnotations');
 
+/**
+ * Computes the number of milliseconds after midnight on the date specified.
+ *
+ * @param dateTime DateTime object to figure out millis from
+ * @returns {number} number of millis in current day
+ */
+exports.computeMillisInCurrentDay = function(e){
+  var msFromMidnight = sundial.getMsFromMidnight(e.time, e.timezoneOffset);
```

Comment:
No, we definitely want to avoid including the `clockDriftOffset` because including it actually effects a *decrease* in our ability to match up to the basal schedules. If you think about anything particular instance where a user adjusted for clock drift, the device times prior to that *included* the drift when the pump was ticking through the basal schedule. If we apply the correction for drift retroactively, we do produce technically more accurate timestamps, but they match up with the schedules less well because we're being smarter than the pump was, if that makes sense.

LMK if you want to talk this through. It's a tricky bit, and I'm probably not doing it justice here.

Tests aren't quite sufficient, but I'll comment on specific inline.